### PR TITLE
update bfgs_trad method

### PR DESCRIPTION
**Pull request by 19hello**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Linked Issue
Fix #...

### Unit Tests and/or Case Tests for my changes
- A unit test is added for each new feature or bug fix.

### What's changed?
- Example: My changes might affect the performance of the application under certain conditions, and I have tested the impact on various scenarios...

### Any changes of core modules? (ignore if not applicable)
- Example: I have added a new virtual function in the esolver base class in order to ...

### CODE REVIEW

File: docs/advanced/opt.md [hunk 1]
```diff
@@ -24,6 +24,10 @@ The [BFGS method](https://en.wikipedia.org/wiki/Broyden%E2%80%93Fletcher%E2%80%9
 
 In ABACUS, we implemented the BFGS method for doing fixed-cell structural relaxation.
 
+### BFGS_TRAD method
+
+The BFGS_TRAD method is an algorithm implemented in ABACUS, referencing the BFGS method from ASE. The previous BFGS method in ABACUS did not perform well for some tests and the BFGS_TRAD method now produces results that are consistant with the BFGS method in ASE. In cases where the previous BFGS method could not converge within a limited number of steps, the BFGS_TRAD method can successfully converge.
```
Comment (dyzheng) [inline docs/advanced/opt.md]: This method is not comes from ASE

File: docs/advanced/input_files/input-main.md [hunk 2]
```diff
@@ -1369,6 +1369,7 @@ These variables are used to control the geometry relaxation.
 - **Description**: The methods to do geometry optimization.
   - cg: using the conjugate gradient (CG) algorithm. Note that there are two implementations of the conjugate gradient (CG) method, see [relax_new](#relax_new).
   - bfgs: using the Broyden–Fletcher–Goldfarb–Shanno (BFGS) algorithm.
+  - bfgs_trad: using BFGS algorithm consistent with the BFGS algorithm in ASE.
```
Comment (dyzheng) [inline docs/advanced/input_files/input-main.md]: same comment.

File: source/module_relax/relax_old/bfgs.cpp [hunk 3]
```diff
@@ -24,7 +24,7 @@ void BFGS::allocate(const int _size) // initialize H0、H、pos0、force0、forc
     steplength = std::vector<double>(size, 0.0);  
 }
 
-void BFGS::relax_step(ModuleBase::matrix _force,UnitCell& ucell) 
+void BFGS::relax_step(const ModuleBase::matrix _force,UnitCell& ucell)
```
Comment (dyzheng) [inline source/module_relax/relax_old/bfgs.cpp]: why not reference?

File: source/module_relax/relax_old/bfgs.h [hunk 4]
```diff
@@ -49,14 +49,14 @@ class BFGS
      * @param _size 
      */
     void allocate(const int _size);//initialize parameters
-    void relax_step(ModuleBase::matrix _force,UnitCell& ucell);//
+    void relax_step(const ModuleBase::matrix _force,UnitCell& ucell);//
```
Comment (dyzheng) [inline source/module_relax/relax_old/bfgs.h]: same question

File: docs/advanced/opt.md (hunk 1 above)
Comment (QuantumMisaka) [reply docs/advanced/opt.md]: @19hello This method may be the BFGS used by ASE, but not from ASE. You need to specify the main algorithm this BFGS and its difference with old BFGS.

File: docs/advanced/opt.md [hunk 5]
```diff
@@ -24,9 +24,9 @@ The [BFGS method](https://en.wikipedia.org/wiki/Broyden%E2%80%93Fletcher%E2%80%9
 
 In ABACUS, we implemented the BFGS method for doing fixed-cell structural relaxation.
 
-### BFGS_TRAD method
+We have alse implemented an alternative BFGS method, which can be called by using the keyword 'bfgs_trad'.
 
-The BFGS_TRAD method is an algorithm implemented in ABACUS, referencing the BFGS method from ASE. The previous BFGS method in ABACUS did not perform well for some tests and the BFGS_TRAD method now produces results that are consistant with the BFGS method in ASE. In cases where the previous BFGS method could not converge within a limited number of steps, the BFGS_TRAD method can successfully converge.
+The bfgs_trad method is a quasi-Newton method that substitute an approximate matrix B for the Hessian matrix. The optimization direction is determined by the inverse of B, therefore, only the inverse of B is iteratively updated and no time-consuming operations such as matrix inversion are involved.
```
Comment (QuantumMisaka) [inline docs/advanced/opt.md]: Users will be curious about the difference with this (so-called 'standard' for common users with not so much knowledge about detailed BFGS algorithm) BFGS and the BFGS used by ABACUS before. will this information be added ?

Comment (QuantumMisaka) [reply docs/advanced/opt.md]: And, a trivial question: why use `trad` but not `std` for abbreviate of standard

**Comment by QuantumMisaka** [comment]
@19hello Any updates ?

**Review by QuantumMisaka — APPROVED** [review]

File: docs/advanced/opt.md (hunk 1 above)
Comment (19hello) [reply docs/advanced/opt.md]: I have modified the introduction of BFGS in the document.

**Review by dyzheng — APPROVED** [review]

**Review by mohanchen — APPROVED** [review]

**Pull request merged**